### PR TITLE
reafactor: Combine most of duplicated code and unify conditional rendering

### PR DESCRIPTION
**Pull request by Mrkazik99**

### CODE REVIEW

**Comment by loboda4450** [comment]
do not even try to assign me as a reviewer XD

**Review by loboda4450 — DISMISSED** [review]

**Review by bouassaba — CHANGES_REQUESTED** [review]
Move `ui/src/types/queries.ts` back inside `@/client`.

**Review by bouassaba — APPROVED** [review]

File: ui/.golangci.yml [hunk 1]
```diff
@@ -2,10 +2,10 @@
 linters:
   enable-all: true
   disable:
-    - exhaustruct    # Structs may be uninitialized
-    - nlreturn       # covered by wsl cuddle rules
+    - exhaustruct # Structs may be uninitialized
```
Comment (dsonck92) [inline ui/.golangci.yml:5]: The formatter got a little zealous I think. I would revert this file as the comment alignment had reasons

Comment (bouassaba) [reply ui/.golangci.yml:5]: I will make a new PR to revert this, and I will add this file to `.prettierignore`.

**Pull request merged**